src/db.rs

```diff
@@ -55,6 +55,8 @@ pub async fn dataset_upsert(
 }
 
 /// Pull a Dataset
+// @coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
```

Comment:
@coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
- Check to make sure we get an error if a non-existent dataset is selected.